### ISSUE TITLE
Embulk >= 0.10.26 raises the exception `java.lang.ClassFormatError: Truncated class file`

### ISSUE THREAD

**Issue by hiroyuki-sato** [embulk/embulk#1408]
Issue Type: Bug Report
-----------------------

* Write the following environmental information.
  * OS version: macOS 11.2.3
  * Java version: 1.8
  * Embulk version: 0.10.28-SNAPSHOT

Embulk >= 0.10.26 raises the exception `java.lang.ClassFormatError: Truncated class file`  in the following case.

### Reproduce steps.

* commit: 73777e4a18495aeb4b5bedbaaedc7bbc12dc13ad
* branch: [upstream/modernize-csv-plugins](https://github.com/embulk/embulk/pull/1399/)

```
embulk example foo
cp foo/csv/sample_01.csv.gz foo/csv/sample_02.csv.gz
cp foo/csv/sample_01.csv.gz foo/csv/sample_03.csv.gz
./build/executable/embulk-0.10.28-SNAPSHOT.jar guess seed.yml -o conf.yml
./build/executable/embulk-0.10.28-SNAPSHOT.jar run conf.yml
```

### log

```
./build/executable/embulk-0.10.28-SNAPSHOT.jar run /tmp/conf.yml
2021-03-17 15:27:34.146 +0900 [INFO] (main): embulk_home is set by the location of embulk.properties found in: /Users/usr/.embulk
2021-03-17 15:27:35.105 +0900 [INFO] (main): Started Embulk v0.10.28-SNAPSHOT
2021-03-17 15:27:35.369 +0900 [INFO] (0001:transaction): Listing local files at directory '/private/tmp/foo/csv' filtering filename by prefix 'sample_'
2021-03-17 15:27:35.369 +0900 [INFO] (0001:transaction): "follow_symlinks" is set false. Note that symbolic links to directories are skipped.
2021-03-17 15:27:35.372 +0900 [INFO] (0001:transaction): Loading files [/private/tmp/foo/csv/sample_02.csv.gz, /private/tmp/foo/csv/sample_01.csv.gz, /private/tmp/foo/csv/sample_03.csv.gz]
2021-03-17 15:27:35.496 +0900 [INFO] (0001:transaction): Using local thread executor with max_threads=8 / output tasks 6 = input tasks 3 * 2
2021-03-17 15:27:35.543 +0900 [INFO] (0001:transaction): {done:  0 / 3, running: 0}
2021-03-17 15:27:35.544 +0900 [INFO] (0001:transaction): {done:  0 / 3, running: 0}
2021-03-17 15:27:35.585 +0900 [INFO] (0001:transaction): {done:  1 / 3, running: 0}
1,32864,2015-01-27 19:23:49,20150127,embulk
2,14824,2015-01-27 19:01:23,20150127,embulk jruby
3,27559,2015-01-28 02:20:02,20150128,Embulk "csv" parser plugin
4,11270,2015-01-29 11:54:36,20150129,
2021-03-17 15:27:35.796 +0900 [INFO] (0001:transaction): {done:  2 / 3, running: 0}
org.embulk.exec.PartialExecutionException: java.lang.RuntimeException: java.lang.ClassFormatError: Truncated class file
	at org.embulk.exec.BulkLoader$LoaderState.buildPartialExecuteException(BulkLoader.java:344)
	at org.embulk.exec.BulkLoader.doRun(BulkLoader.java:584)
	at org.embulk.exec.BulkLoader.access$000(BulkLoader.java:37)
	at org.embulk.exec.BulkLoader$1.run(BulkLoader.java:357)
	at org.embulk.exec.BulkLoader$1.run(BulkLoader.java:354)
	at org.embulk.spi.ExecInternal.doWith(ExecInternal.java:26)
	at org.embulk.exec.BulkLoader.run(BulkLoader.java:354)
	at org.embulk.EmbulkEmbed.run(EmbulkEmbed.java:324)
	at org.embulk.EmbulkRunner.runInternal(EmbulkRunner.java:288)
	at org.embulk.EmbulkRunner.run(EmbulkRunner.java:153)
	at org.embulk.cli.EmbulkRun.runInternal(EmbulkRun.java:119)
	at org.embulk.cli.EmbulkRun.run(EmbulkRun.java:24)
	at org.embulk.cli.Main.main(Main.java:55)
Caused by: java.lang.RuntimeException: java.lang.ClassFormatError: Truncated class file
	at org.embulk.exec.BulkLoader$LoaderState.getRepresentativeException(BulkLoader.java:289)
	at org.embulk.exec.BulkLoader.execute(BulkLoader.java:702)
	at org.embulk.exec.BulkLoader.access$300(BulkLoader.java:37)
	at org.embulk.exec.BulkLoader$4$1$1$1.run(BulkLoader.java:546)
	at org.embulk.output.stdout.StdoutOutputPlugin.resume(StdoutOutputPlugin.java:71)
	at org.embulk.output.stdout.StdoutOutputPlugin.transaction(StdoutOutputPlugin.java:64)
	at org.embulk.exec.BulkLoader$4$1$1.transaction(BulkLoader.java:539)
	at org.embulk.exec.LocalExecutorPlugin.transaction(LocalExecutorPlugin.java:51)
	at org.embulk.exec.BulkLoader$4$1.run(BulkLoader.java:534)
	at org.embulk.spi.util.FiltersInternal$RecursiveControl.transaction(FiltersInternal.java:84)
	at org.embulk.spi.util.FiltersInternal.transaction(FiltersInternal.java:42)
	at org.embulk.exec.BulkLoader$4.run(BulkLoader.java:529)
	at org.embulk.spi.FileInputRunner$RunnerControl$1$1.run(FileInputRunner.java:111)
	at org.embulk.parser.csv.CsvParserPlugin.transaction(CsvParserPlugin.java:309)
	at org.embulk.spi.FileInputRunner$RunnerControl$1.run(FileInputRunner.java:106)
	at org.embulk.spi.util.DecodersInternal$RecursiveControl.transaction(DecodersInternal.java:66)
	at org.embulk.spi.util.DecodersInternal$RecursiveControl$1.run(DecodersInternal.java:62)
	at org.embulk.decoder.gzip.GzipFileDecoderPlugin.transaction(GzipFileDecoderPlugin.java:38)
	at org.embulk.spi.util.DecodersInternal$RecursiveControl.transaction(DecodersInternal.java:58)
	at org.embulk.spi.util.DecodersInternal.transaction(DecodersInternal.java:28)
	at org.embulk.spi.FileInputRunner$RunnerControl.run(FileInputRunner.java:104)
	at org.embulk.input.file.LocalFileInputPlugin.resume(LocalFileInputPlugin.java:92)
	at org.embulk.input.file.LocalFileInputPlugin.transaction(LocalFileInputPlugin.java:85)
	at org.embulk.spi.FileInputRunner.transaction(FileInputRunner.java:61)
	at org.embulk.exec.BulkLoader.doRun(BulkLoader.java:525)
	... 11 more
	Suppressed: java.lang.ClassFormatError: Truncated class file
		at java.lang.ClassLoader.defineClass1(Native Method)
		at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
		at java.lang.ClassLoader.defineClass(ClassLoader.java:839)
		at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
		at org.embulk.deps.SelfContainedJarAwareURLClassLoader.defineClassFromEmbulkSelfContainedJarFiles(SelfContainedJarAwareURLClassLoader.java:217)
		at org.embulk.deps.SelfContainedJarAwareURLClassLoader.access$000(SelfContainedJarAwareURLClassLoader.java:30)
		at org.embulk.deps.SelfContainedJarAwareURLClassLoader$1.run(SelfContainedJarAwareURLClassLoader.java:60)
		at org.embulk.deps.SelfContainedJarAwareURLClassLoader$1.run(SelfContainedJarAwareURLClassLoader.java:56)
		at java.security.AccessController.doPrivileged(Native Method)
		at org.embulk.deps.SelfContainedJarAwareURLClassLoader.findClass(SelfContainedJarAwareURLClassLoader.java:55)
		at org.embulk.plugin.PluginClassLoader.loadClass(PluginClassLoader.java:138)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
		at com.fasterxml.jackson.datatype.jdk8.Jdk8Module.setupModule(Jdk8Module.java:25)
		at com.fasterxml.jackson.databind.ObjectMapper.registerModule(ObjectMapper.java:718)
		at org.embulk.util.config.ConfigMapperFactory.mapperForTask(ConfigMapperFactory.java:231)
		at org.embulk.util.config.ConfigMapperFactory.createTaskMapper(ConfigMapperFactory.java:172)
		at org.embulk.output.stdout.StdoutOutputPlugin.open(StdoutOutputPlugin.java:82)
		at org.embulk.exec.LocalExecutorPlugin$ScatterTransactionalPageOutput.openOutputs(LocalExecutorPlugin.java:402)
		at org.embulk.exec.LocalExecutorPlugin$ScatterExecutor.runInputTask(LocalExecutorPlugin.java:257)
		at org.embulk.exec.LocalExecutorPlugin$ScatterExecutor.access$100(LocalExecutorPlugin.java:195)
		at org.embulk.exec.LocalExecutorPlugin$ScatterExecutor$1.call(LocalExecutorPlugin.java:234)
		at org.embulk.exec.LocalExecutorPlugin$ScatterExecutor$1.call(LocalExecutorPlugin.java:231)
		at java.util.concurrent.FutureTask.run(FutureTask.java:266)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
		at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassFormatError: Truncated class file
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:839)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
	at org.embulk.deps.SelfContainedJarAwareURLClassLoader.defineClassFromEmbulkSelfContainedJarFiles(SelfContainedJarAwareURLClassLoader.java:217)
	at org.embulk.deps.SelfContainedJarAwareURLClassLoader.access$000(SelfContainedJarAwareURLClassLoader.java:30)
	at org.embulk.deps.SelfContainedJarAwareURLClassLoader$1.run(SelfContainedJarAwareURLClassLoader.java:60)
	at org.embulk.deps.SelfContainedJarAwareURLClassLoader$1.run(SelfContainedJarAwareURLClassLoader.java:56)
	at java.security.AccessController.doPrivileged(Native Method)
	at org.embulk.deps.SelfContainedJarAwareURLClassLoader.findClass(SelfContainedJarAwareURLClassLoader.java:55)
	at org.embulk.plugin.PluginClassLoader.loadClass(PluginClassLoader.java:159)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at org.embulk.plugin.jar.JarPluginLoader.loadSelfContainedJarPluginMainClass(JarPluginLoader.java:188)
	at org.embulk.plugin.jar.JarPluginLoader.loadSelfContained(JarPluginLoader.java:64)
	at org.embulk.plugin.SelfContainedPluginSource.newPlugin(SelfContainedPluginSource.java:79)
	at org.embulk.plugin.PluginManager.newPlugin(PluginManager.java:85)
	at org.embulk.spi.ExecSessionInternal.newPlugin(ExecSessionInternal.java:323)
	at org.embulk.exec.LocalExecutorPlugin$ScatterExecutor.runInputTask(LocalExecutorPlugin.java:252)
	at org.embulk.exec.LocalExecutorPlugin$ScatterExecutor.access$100(LocalExecutorPlugin.java:195)
	at org.embulk.exec.LocalExecutorPlugin$ScatterExecutor$1.call(LocalExecutorPlugin.java:234)
	at org.embulk.exec.LocalExecutorPlugin$ScatterExecutor$1.call(LocalExecutorPlugin.java:231)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

Error: java.lang.RuntimeException: java.lang.ClassFormatError: Truncated class file
```

## Reference

* [Twitter](https://twitter.com/hiroysato/status/1371101685741879297?s=20)

**Comment by dmikurube** [comment]
Thanks for the reproducing case...! We're going to have a deeper look.

**Comment by dmikurube** [comment]
Should be fixed by #1411 in v0.10.28.
